hane-kernel/src/term.rs

```diff
@@ -440,10 +438,8 @@ impl<M: Clone, B: Clone> Term<M, B> {
                 x_sort
                     .expect_sort(global, lenv)
                     .map_err(|err| (x_tp.meta.clone(), err))?;
-                let t_tp = {
-                    let mut lenv = lenv.push(Entry::new(x.clone(), x_tp.clone()));
-                    t.type_check(global, &mut lenv)?
-                };
+                let mut lenv = lenv.push(Entry::new(x.clone(), x_tp.clone()));
```

Comment:
Why not scope `lenv`?

Reply:
Why. It doesn't hurt to have it live an extra line. Scoping it would however make the code less readable.